### PR TITLE
Closes #449 | Add/Update Dataloader Thai LOTUS

### LINKED ISSUE #449: Create dataset loader for AIFORTHAI - LotusCorpus
Dataloader name: `tha_lotus/tha_lotus.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?tha_lotus

|     Dataset| tha_lotus |
|-------------|---|
| Description | The Large vOcabualry Thai continUous Speech recognition (LOTUS) corpus was designed for developing large vocabulary continuous speech recognition (LVCSR), spoken dialogue system, speech dictation, broadcast news transcriber. It contains two datasets, one for training acoustic model, another for training a language model. |
| Subsets | - |
| Languages | tha |
| Tasks | Automatic Speech Recognition |
| License | Creative Commons Attribution Non Commercial Share Alike 3.0 (cc-by-nc-sa-3.0) |
| Homepage | https://github.com/korakot/corpus/releases/download/v1.0/AIFORTHAI-LotusCorpus.zip |
| HF URL | - |
| Paper URL | https://doi.org/10.1109/ICSDA.2009.5278377 |

### PR DESCRIPTION
**Pull request by sabilmakbar**
Closes #449

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/tha_lotus/tha_lotus.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with:
1. `python -m tests.test_seacrowd seacrowd/sea_datasets/tha_lotus/tha_lotus.py --subset_id tha_lotus_closetalk_clean`
2. `python -m tests.test_seacrowd seacrowd/sea_datasets/tha_lotus/tha_lotus.py --subset_id tha_lotus_closetalk_office`
3. `python -m tests.test_seacrowd seacrowd/sea_datasets/tha_lotus/tha_lotus.py --subset_id tha_lotus_unidrection_clean`
4. `python -m tests.test_seacrowd seacrowd/sea_datasets/tha_lotus/tha_lotus.py --subset_id tha_lotus_unidrection_office`
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/tha_lotus/tha_lotus.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+_DATASETNAME = "tha_lotus"
+_DESCRIPTION = r"""
+The Large vOcabualry Thai continUous Speech recognition (LOTUS) corpus was designed for developing large vocabulary
+continuous speech recognition (LVCSR), spoken dialogue system, speech dictation, broadcast news transcriber.
+It contains two datasets, one for training acoustic model, another for training a language model.
+"""
+
+_HOMEPAGE = "https://github.com/korakot/corpus/tree/main/LOTUS"
+_LICENSE = Licenses.CC.value
```
Comment (holylovenia) [inline seacrowd/sea_datasets/tha_lotus/tha_lotus.py:44]: Could you please change the license to Creative Commons Attribution Non Commercial Share Alike 3.0 (cc-by-nc-sa-3.0)?

**Review by holylovenia — APPROVED** [review]
Good work as always, @sabilmakbar! I have only one minor suggestion below. Let's wait for @faridlazuarda's review.

File: seacrowd/sea_datasets/tha_lotus/tha_lotus.py (hunk 1 above)
Comment (sabilmakbar) [reply seacrowd/sea_datasets/tha_lotus/tha_lotus.py]: thanks @holylovenia!~ missed this detail

**Review by faridlazuarda — COMMENTED** [review]
```
python -m tests.test_seacrowd seacrowd/sea_datasets/tha_lotus/tha_lotus.py --subset_id tha_lotus_closetalk_clean

Traceback (most recent call last):
  File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/tests/test_seacrowd.py", line 14, in <module>
    from seacrowd.utils.constants import Tasks, TASK_TO_SCHEMA, VALID_TASKS, VALID_SCHEMAS, SCHEMA_TO_FEATURES, TASK_TO_FEATURES
  File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/__init__.py", line 1, in <module>
    from .utils.constants import Tasks
  File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/utils/constants.py", line 40, in <module>
    class Tasks(Enum):
  File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/utils/constants.py", line 131, in Tasks
    OPTICAL_CHARACTER_RECOGNITION = "OCR"
  File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/enum.py", line 134, in __setitem__
    raise TypeError('Attempted to reuse key: %r' % key)
TypeError: Attempted to reuse key: 'OPTICAL_CHARACTER_RECOGNITION'
```

Hello, any idea what is the cause for this error? @sabilmakbar @holylovenia

**Comment by holylovenia** [comment]
> ```
> python -m tests.test_seacrowd seacrowd/sea_datasets/tha_lotus/tha_lotus.py --subset_id tha_lotus_closetalk_clean
> 
> Traceback (most recent call last):
>   File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/runpy.py", line 196, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/runpy.py", line 86, in _run_code
>     exec(code, run_globals)
>   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/tests/test_seacrowd.py", line 14, in <module>
>     from seacrowd.utils.constants import Tasks, TASK_TO_SCHEMA, VALID_TASKS, VALID_SCHEMAS, SCHEMA_TO_FEATURES, TASK_TO_FEATURES
>   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/__init__.py", line 1, in <module>
>     from .utils.constants import Tasks
>   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/utils/constants.py", line 40, in <module>
>     class Tasks(Enum):
>   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/utils/constants.py", line 131, in Tasks
>     OPTICAL_CHARACTER_RECOGNITION = "OCR"
>   File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/enum.py", line 134, in __setitem__
>     raise TypeError('Attempted to reuse key: %r' % key)
> TypeError: Attempted to reuse key: 'OPTICAL_CHARACTER_RECOGNITION'
> ```
> 
> Hello, any idea what is the cause for this error? @sabilmakbar @holylovenia

Can you pull from `master`, @faridlazuarda? Previously `constants.py` had duplicated lines for `OPTICAL_CHARACTER_RECOGNITION`, that was what triggered the error for me.

**Review by faridlazuarda — APPROVED** [review]
> > ```
> > python -m tests.test_seacrowd seacrowd/sea_datasets/tha_lotus/tha_lotus.py --subset_id tha_lotus_closetalk_clean
> > 
> > Traceback (most recent call last):
> >   File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/runpy.py", line 196, in _run_module_as_main
> >     return _run_code(code, main_globals, None,
> >   File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/runpy.py", line 86, in _run_code
> >     exec(code, run_globals)
> >   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/tests/test_seacrowd.py", line 14, in <module>
> >     from seacrowd.utils.constants import Tasks, TASK_TO_SCHEMA, VALID_TASKS, VALID_SCHEMAS, SCHEMA_TO_FEATURES, TASK_TO_FEATURES
> >   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/__init__.py", line 1, in <module>
> >     from .utils.constants import Tasks
> >   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/utils/constants.py", line 40, in <module>
> >     class Tasks(Enum):
> >   File "/Users/faridadilazuarda/Documents/GitHub/seacrowd-datahub/seacrowd/utils/constants.py", line 131, in Tasks
> >     OPTICAL_CHARACTER_RECOGNITION = "OCR"
> >   File "/Users/faridadilazuarda/miniconda3/envs/env-seacrowd/lib/python3.10/enum.py", line 134, in __setitem__
> >     raise TypeError('Attempted to reuse key: %r' % key)
> > TypeError: Attempted to reuse key: 'OPTICAL_CHARACTER_RECOGNITION'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Hello, any idea what is the cause for this error? @sabilmakbar @holylovenia
> 
> Can you pull from `master` and see if the error persists or not, @faridlazuarda? Previously `constants.py` had duplicated lines for `OPTICAL_CHARACTER_RECOGNITION`, that was what triggered the error for me.


Yes, this was the issue, thank you! @holylovenia. Now I can run the code perfectly. LGTM!

**Comment by sabilmakbar** [comment]
okay, thanks for the reviews, @holylovenia and @faridlazuarda

**Pull request merged**